tests/update/patch_sle.pm

```diff
@@ -44,6 +44,7 @@ sub patching_sle {
         zypper_call "lr";
         zypper_call "mr --disable --all";
         save_screenshot;
+        set_scc_proxy_url;    # Set SCC_PROXY_URL if needed
```

Comment:
In fact, I have the same ticket to fix and a test log already got. 
http://10.161.8.44/tests/39#
Normally, we migration team use SCC_PROXY_URL for online migration, while use SCC_URL for offline migration. Besides, here we need condition to only permit base system equal current developing system such as SLE12SP5 to use proxy scc to do fully update on base system. WDYT?

Please have a look on my code here:
if ((my $u = get_var('SCC_URL')) && (get_var("HDDVERSION") =~ get_var("VERSION")) && is_upgrade()) {
        type_string "echo 'url: $u' > /etc/SUSEConnect\n";
    }
    save_screenshot;

    sle_register("register");

Other modification are OK for me.

Reply:
I checked and your code is not working for 12-SPx upgrade to 12-SP5, in that case the value of `SCC_URL` is added in `/etc/SUSEConnect` but that file should not be changed: http://1b147.qa.suse.de/tests/5577

By default `SCC_URL` is set to the internal SCC Proxy, it's not related to offline or online upgrade (but it can be used for this as I do :-)).

I've search for something else but I didn't find a better solution at this time.

Reply:
In fact, for 12-SPx to 12-SP5 we don't need to register Proxy SCC, while register to SCC. We need only focus on the SLE-N-1(SLE12SP5) to SLE-N(SLE15SP1), and for this scenario we need to use Proxy SCC to do fully update on base system.
You can continue use the SCC_PROXY_URL, since this is special scenario and won't affect the using of SCC_URL or SCC_PROXY_URL for migraion test.
About the invoke condition, I try to limit this only for the scenario SLE-N-1(SLE12SP5) to SLE-N(SLE15SP1) as (get_var("HDDVERSION") =~ get_var("VERSION")) and only for migraion test as is_upgrade().

Reply:
> In fact, for 12-SPx to 12-SP5 we don't need to register Proxy SCC, while register to SCC. We need only focus on the SLE-N-1(SLE12SP5) to SLE-N(SLE15SP1)

Agree, this is why the function `set_scc_proxy_url` will only be used if `SCC_PROXY_URL` is set (as done previously), I just moved the call to the function in another step.

I think that all should be good, but if you have a case where my modifications are not OK please point it to me, I will try to check and modify my code to handle all cases. Your suggested modification is not working at all (I tested it) and I honestly don't know how it could work.

Reply:
I just want to limit the set_scc_proxy_url for migration test case for SLE-N-1(SLE12SP5) to SLE-N(SLE15SP1). I have checked your setting, variable already been 'SLE15-SP1' so (get_var("HDDVERSION") =~ get_var("VERSION")) not work any more. So just set condition as is_upgrade() is OK.   I will update code for setting the 'VERSION' as SLE15SP1 in PR#https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/7624 later.

Reply:
Set_scc_proxy_url here without condition to limit will cause some cases wrong, such as migration from SLE12SP2 to SLE12SP5 will update from the the proxy SCC of SLE12SP5 while this should update from SCC if test with variable SCC_PROXY_URL.
We can campare 'HDDVERSION' and 'UPGRADE_TARGET_VERSION' to check if it is the scenario of SLE-N-1 to SLE-N such as SLE12SP5. But you have changed the logic of UPGRADE_TARGET_VERSION, so it is not the original target any more.  I think you can add another variable(ex: ORIGINAL_TARGET_VERSION) to save the original target after 'set_var('UPGRADE_TARGET_VERSION', get_var('VERSION')) if (!get_var('UPGRADE_TARGET_VERSION'));'.
Then
set_scc_proxy_url if  ((check_var("HDDVERSION", get_var("ORIGINAL_TARGET_VERSION")) && is_upgrade());
WDYT?

Reply:
I tested your last suggestion and it seems to work: http://1b147.qa.suse.de/tests/5611 and http://1b147.qa.suse.de/tests/5612 (the failure is not related to the upgrade code, but a [product bug](https://bugzilla.suse.com/show_bug.cgi?id=1138865)).